pkg/phases/kernel.go

```diff
@@ -0,0 +1,164 @@
+package phases
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/flanksource/konfigadm/pkg/types"
+	"github.com/flanksource/konfigadm/pkg/utils"
+)
+
+var Kernel types.AllPhases = kernel{}
+
+type kernel struct{}
+
+func (p kernel) ApplyPhase(sys *types.Config, ctx *types.SystemContext) ([]types.Command, types.Filesystem, error) {
+	commands := types.Commands{}
+	files := types.Filesystem{}
+	var os OS
+	var err error
+	if len(*sys.Kernel) == 0 {
+		return []types.Command{}, files, nil
+	}
+	// Ensure repo caches are up to date before trying to install kernel
+	for _, os := range SupportedOperatingSystems {
+		commands = *commands.Append(os.GetPackageManager().Update().WithTags(os.GetTags()...))
+	}
+	for _, kern := range *sys.Kernel {
+		var oslist []OS
+		if len(kern.Flags) > 0 {
+			// Kernel version is tagged for a specific OS (family)
+			// Add specific kernel and headers package
+			for _, flag := range kern.Flags {
+				if os, err = GetOSForTag(flag); err != nil {
+					continue
+				}
+				oslist = append(oslist, os)
+			}
+		} else {
+			// Input is not tagged
+			// Add for all supported OSes
+			oslist = SupportedOperatingSystems
+		}
+		for _, os := range oslist {
+			commandtags := os.GetTags()
+			// rpm based systems will fail to install kernels in containers
```

Comment:
I think move this comment into the test file, and we can then remove this Fedora filter